modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -377,6 +380,7 @@ public String toParamName(String name) {
 
     @Override
     public String toModelName(String name) {
+        name = super.toModelName(name);
```

Comment:
I think removing the prefix/suffix after the super method added it is not a good idea (if you don't want them, simply don't pass those parameters), but there might be the use case of using a different separator/naming convention (instead of the camelCase) for a programming language, which could be a reason to implement it in the subclass too.

So I would let it stay in the superclass method (which makes it available for all generators which don't override it), but implement it again in the JavaClientCodegen instead of calling `super.toModelName(...)`.